packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -726,11 +726,20 @@ function detachFiber(current: Fiber) {
   // get GC:ed but we don't know which for sure which parent is the current
   // one so we'll settle for GC:ing the subtree of this child. This child
   // itself will be GC:ed when the parent updates the next time.
+  // We do not null out the 'nextEffect' field as it causes tests to fail.
```

Comment:
Do we have a better understanding of this? "Causes tests to fail" doesn't say anything about the intent and whether those tests _should_ fail or if there's some other mistake.

This feels dangerous to me if we don't understand it. How do you know clearing `firstEffect` is safe if `nextEffect` is unsafe? Maybe we just accidentally hit some cases in our tests but not others.

Reply:
In general I'd prefer that we don't use vague wording like "causes tests to fail" in the comments. Either we know why it fails and explain the conceptual reason. Or we don't know, and should understand before merging.

Reply:
@gaearon This PR isn't in a state to be merged. I was more asking @acdlite for some guidance on this, thus my comments.